### ISSUE TITLE
Update IODA use cases to automatically obtain IODA file from JCSDA

### ISSUE THREAD

**Issue by DanielAdriaansen** [dtcenter/METplus#2393]
## Describe the Enhancement ##
There are two current IODA use cases:
https://metplus.readthedocs.io/en/latest/generated/model_applications/data_assimilation/StatAnalysis_fcstHAFS_obsPrepBufr_JEDI_IODA_interface.html#sphx-glr-generated-model-applications-data-assimilation-statanalysis-fcsthafs-obsprepbufr-jedi-ioda-interface-py

https://metplus.readthedocs.io/en/latest/generated/model_applications/data_assimilation/StatAnalysis_fcstGFS_HofX_obsIODAv2_PyEmbed.html#sphx-glr-generated-model-applications-data-assimilation-statanalysis-fcstgfs-hofx-obsiodav2-pyembed-py

This issue is to:
1) Determine whether multiple "versions" of IODA need to be supported, or whether a single use case for the latest/current version of IODA is sufficient
2) Modify the IODAv2 use case to ingest the latest IODA file from JCSDA automatically during CI, which will capture any errors using the new IODA file with CI and prompt changes

### Time Estimate ###
TBD

### Sub-Issues ###
Consider breaking the enhancement down into sub-issues.
- [ ] *Add a checkbox for each sub-issue here.*

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
*Define the source of funding and account keys here or state NONE.*

### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Select **Repository** and/or **Organization** level **Project(s)** or add **alert: NEED CYCLE ASSIGNMENT** label
- [ ] Select **Milestone** as the next official version or **Future Versions**

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdataio](https://github.com/dtcenter/METdataio/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Enhancement Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **develop**.
Branch name: `feature_<Issue Number>_<Description>`
- [ ] Complete the development and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **develop**.
Pull request: `feature <Issue Number> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Development** issues
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Close this issue.

**Comment by jeromebarre** [comment]
@climbfuji As discussed internally before, we need to generate an IODA file example from our CI testing and either store it in an accessible location or push it directly to DTC.

**Comment by jeromebarre** [comment]
ftp push option here:
https://github.com/dtcenter/METplus/discussions/954

**Comment by climbfuji** [comment]
I think I can make ftp work, although S3 is a nicer option. @jeromebarre Please provide me with the name of an output file and the experiment that generates it (ideally one that already runs in CI nightly, see https://github.com/JCSDA-internal/skylab/blob/develop/.github/workflows/run_ec2_pcluster.yaml). Thanks!

**Comment by DanielAdriaansen** [comment]
Here is the current IODAv2 file we are using: https://dtcenter.ucar.edu/dfiles/code/METplus/METplus_Data/v6.0/model_applications/data_assimilation/StatAnalysis_fcstGFS_HofX_obsIODAv2_PyEmbed/sample_hofx_output_sondes.nc4

**Comment by jeromebarre** [comment]
We would need to provide the `fb` IODA files (there should normally be 3) that are produced by `gfs-3dfgat-c12` once the `variational` task is complete.

**Comment by jeromebarre** [comment]
> I think I can make ftp work, although S3 is a nicer option. @jeromebarre Please provide me with the name of an output file and the experiment that generates it (ideally one that already runs in CI nightly, see https://github.com/JCSDA-internal/skylab/blob/develop/.github/workflows/run_ec2_pcluster.yaml). Thanks!

We just had a meeting and we ruled out the FTP option and S3 bucket is the better option.

**Comment by climbfuji** [comment]
@jeromebarre I need to know the exact names of the files and their locations within the experiment. Is it these? Which one of those do you want to upload to an S3 bucket (easiest is all of them)?
```
(venv) ubuntu@ip-10-0-1-189:~/skylab/manual_run$ ls -lart workdir/f72620/20201215T000000Z/obs/
total 19736
-rw-r--r-- 1 ubuntu ubuntu    68495 Nov  6 16:36 obs.AMSUA_N19.20201215T000000Z.nc4
-rw-r--r-- 1 ubuntu ubuntu   314639 Nov  6 16:36 obs.Sondes.20201215T000000Z.nc4
-rw-r--r-- 1 ubuntu ubuntu   379411 Nov  6 16:36 obs.Aircraft.20201215T000000Z.nc4
-rw-rw-r-- 1 ubuntu ubuntu  4262627 Nov  6 16:38 fb.Aircraft.20201215T000000Z.nc4
-rw-rw-r-- 1 ubuntu ubuntu 10238866 Nov  6 16:38 fb.AMSUA_N19.20201215T000000Z.nc4
drwxrwxr-x 2 ubuntu ubuntu     4096 Nov  6 16:38 .
-rw-rw-r-- 1 ubuntu ubuntu  4900380 Nov  6 16:38 fb.Sondes.20201215T000000Z.nc4
drwxrwxr-x 4 ubuntu ubuntu    28672 Nov  6 16:38 ..
```

**Comment by climbfuji** [comment]
@jeromebarre Can you please point me to the corresponding issue in JCSDA/JCSDA-internal? Thanks.

**Comment by climbfuji** [comment]
This has been added, please see https://github.com/JCSDA-internal/skylab/pull/191 for the PR (if you can). The data is available at s3://jedi-test-files/ci/met-ioda

**Comment by DanielAdriaansen** [comment]
@climbfuji I cannot view the PR, and I am also having difficulty accessing the S3 bucket. Is it public? If not, let me know how you want to handle that.

**Comment by climbfuji** [comment]
> @climbfuji I cannot view the PR, and I am also having difficulty accessing the S3 bucket. Is it public? If not, let me know how you want to handle that.

It's public, but I think the easiest way for you to access the data is via http (curl, wget):
```
https://jedi-test-files.s3.amazonaws.com/ci/met-ioda/fb.Aircraft.20201215T000000Z.nc4
https://jedi-test-files.s3.amazonaws.com/ci/met-ioda/fb.AMSUA_N19.20201215T000000Z.nc4
https://jedi-test-files.s3.amazonaws.com/ci/met-ioda/fb.Sondes.20201215T000000Z.nc4
```

**Comment by DanielAdriaansen** [comment]
Thanks! We were able to access and open the files.

**Comment by DanielAdriaansen** [comment]
Greetings @climbfuji @jeromebarre. Just wanted to see if the IODA files at the AWS links are still being updated? We have some support to work in this area again, and are going to consider moving this forward a bit more, so wanted to see whether these were still current or not, and if they are current which version(s) of IODA they will apply to. Thanks!

**Comment by climbfuji** [comment]
Adding @eap as the new JCSDA infrastructure team lead. These files are (were?) generated by the Skylab CI runs that are run on schedule every night. Part of the workflow is (was?) uploading the files that were produced during the run to S3.

**Comment by fabiolrdiniz** [comment]
Thanks for pointing this out, @climbfuji. I just checked the files are still uploaded to the S3 bucket. The following is from last night's CI run (https://github.com/JCSDA-internal/skylab/actions/runs/11606925918/job/32319656989):

```console
Uploading IODA test files to S3 bucket
Completed 256.0 KiB/2.7 MiB (3.6 MiB/s) with 6 file(s) remaining
Completed 512.0 KiB/2.7 MiB (7.2 MiB/s) with 6 file(s) remaining
Completed 768.0 KiB/2.7 MiB (8.6 MiB/s) with 6 file(s) remaining
Completed 1.0 MiB/2.7 MiB (11.3 MiB/s) with 6 file(s) remaining 
Completed 1.2 MiB/2.7 MiB (9.8 MiB/s) with 6 file(s) remaining  
Completed 1.5 MiB/2.7 MiB (11.6 MiB/s) with 6 file(s) remaining 
Completed 1.5 MiB/2.7 MiB (11.9 MiB/s) with 6 file(s) remaining 
upload: workdir/71c948/20201215T000000Z/obs/fb.Aircraft.20201215T000000Z.nc4 to s3://jedi-test-files/ci/met-ioda/fb.Aircraft.20201215T000000Z.nc4
Completed 1.5 MiB/2.7 MiB (11.9 MiB/s) with 5 file(s) remaining
Completed 1.8 MiB/2.7 MiB (13.6 MiB/s) with 5 file(s) remaining
Completed 2.0 MiB/2.7 MiB (14.5 MiB/s) with 5 file(s) remaining
Completed 2.3 MiB/2.7 MiB (12.7 MiB/s) with 5 file(s) remaining
upload: workdir/71c948/20201215T000000Z/obs/fb.AMSUA_N19.20201215T000000Z.nc4 to s3://jedi-test-files/ci/met-ioda/fb.AMSUA_N19.20201215T000000Z.nc4
Completed 2.3 MiB/2.7 MiB (12.7 MiB/s) with 4 file(s) remaining
Completed 2.3 MiB/2.7 MiB (11.1 MiB/s) with 4 file(s) remaining
upload: workdir/71c948/20201215T000000Z/obs/obs.Sondes.20201215T000000Z.nc4 to s3://jedi-test-files/ci/met-ioda/obs.Sondes.20201215T000000Z.nc4
Completed 2.3 MiB/2.7 MiB (11.1 MiB/s) with 3 file(s) remaining
Completed 2.5 MiB/2.7 MiB (10.3 MiB/s) with 3 file(s) remaining
upload: workdir/71c948/20201215T000000Z/obs/fb.Sondes.20201215T000000Z.nc4 to s3://jedi-test-files/ci/met-ioda/fb.Sondes.20201215T000000Z.nc4
Completed 2.5 MiB/2.7 MiB (10.3 MiB/s) with 2 file(s) remaining
Completed 2.6 MiB/2.7 MiB (10.3 MiB/s) with 2 file(s) remaining
upload: workdir/71c948/20201215T000000Z/obs/obs.Aircraft.20201215T000000Z.nc4 to s3://jedi-test-files/ci/met-ioda/obs.Aircraft.20201215T000000Z.nc4
Completed 2.6 MiB/2.7 MiB (10.3 MiB/s) with 1 file(s) remaining
Completed 2.7 MiB/2.7 MiB (10.3 MiB/s) with 1 file(s) remaining
upload: workdir/71c948/20201215T000000Z/obs/obs.AMSUA_N19.20201215T000000Z.nc4 to s3://jedi-test-files/ci/met-ioda/obs.AMSUA_N19.20201215T000000Z.nc4
+ rsync -av /home/ubuntu/skylab/jedi_run/workdir/71c948/20201215T000000Z/obs/ /home/ubuntu/simobs/input_from_skylab_ci/ci/
sending incremental file list
./
fb.AMSUA_N19.20201215T000000Z.nc4
fb.Aircraft.20201215T000000Z.nc4
fb.Sondes.20201215T000000Z.nc4
obs.AMSUA_N19.20201215T000000Z.nc4
obs.Aircraft.20201215T000000Z.nc4
obs.Sondes.20201215T000000Z.nc4

sent 2,812,708 bytes  received 133 bytes  5,625,682.00 bytes/sec
total size is 2,811,470  speedup is 1.00
```
